src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -191,21 +191,80 @@ def load_arguments(self, _):
+    with self.argument_context('deployment operation') as c:
         c.argument('operation_ids', nargs='+', help='A list of operation ids to show')
 
+    with self.argument_context('deployment sub') as c:
+        c.argument('deployment_location', arg_type=get_location_type(self.cli_ctx), required=True)
+
+    with self.argument_context('deployment sub create') as c:
+        c.argument('deployment_name', options_list=['--name', '-n'], required=False,
+                   help='The deployment name. Default to template file base name')
+        c.argument('handle_extended_json_format', arg_type=extended_json_format_type)
+
+    with self.argument_context('deployment sub validate') as c:
+        c.argument('deployment_name', options_list=['--name', '-n'], required=False,
+                   help='The deployment name. Default to template file base name')
+        c.argument('handle_extended_json_format', arg_type=extended_json_format_type)
+
+    with self.argument_context('deployment group') as c:
+        c.argument('resource_group_name', arg_type=resource_group_name_type, completer=get_resource_group_completion_list, required=True)
+        c.argument('mode', arg_type=get_enum_type(DeploymentMode, default='incremental'), help='Incremental (only add resources to resource group) or Complete (remove extra resources from resource group)')
+
+    with self.argument_context('deployment group create') as c:
+        c.argument('deployment_name', options_list=['--name', '-n'], required=False,
+                   help='The deployment name. Default to template file base name')
+        c.argument('handle_extended_json_format', arg_type=extended_json_format_type)
+
+    with self.argument_context('deployment group validate') as c:
+        c.argument('deployment_name', options_list=['--name', '-n'], required=False,
+                   help='The deployment name. Default to template file base name')
+        c.argument('handle_extended_json_format', arg_type=extended_json_format_type)
+
+    with self.argument_context('deployment mg') as c:
+        c.argument('management_group_id', options_list=['--management-group-id', '-m'], required=True, help='The management group id.')
+        c.argument('deployment_location', arg_type=get_location_type(self.cli_ctx), required=True)
+
+    with self.argument_context('deployment mg create') as c:
+        c.argument('deployment_name', options_list=['--name', '-n'], required=False,
+                   help='The deployment name. Default to template file base name')
+        c.argument('handle_extended_json_format', arg_type=extended_json_format_type)
+
+    with self.argument_context('deployment mg validate') as c:
+        c.argument('deployment_name', options_list=['--name', '-n'], required=False,
+                   help='The deployment name. Default to template file base name')
+        c.argument('handle_extended_json_format', arg_type=extended_json_format_type)
```

Comment:
Maybe you can use CLIArgumentType to define your argument, so that you can avoid duplicates.
E.g.
```
ultra_ssd_enabled_type = CLIArgumentType(
        arg_type=get_three_state_flag(), min_api='2018-06-01',
        help='Enables or disables the capability to have 1 or more managed data disks with UltraSSD_LRS storage account')
```

Reply:
sure I can update that.